### PR TITLE
Fixed getExecutionPlanDescription on compiled results

### PR DESCRIPTION
**Pull request by SaschaPeukert**

### CODE REVIEW

File: enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/ExecutionResultAcceptanceTest.scala [hunk 1]
```diff
@@ -38,4 +41,40 @@ class ExecutionResultAcceptanceTest extends ExecutionEngineFunSuite{
...
     })
   }
+
+  test("without PROFILE it shouldn't be needed to iterate over the results before calling getExecutionPlanDescription in compiled runtime") {
+    val query = "MATCH (n) WHERE n.prop = 1 RETURN n"
+
+    val runtime = CompiledRuntimeOption.name
+    val description1 = executeQueryAndGetExecutionPlanDescription(query, runtime, false)
+    val description2 = executeQueryAndGetExecutionPlanDescription(query, runtime, true)
+    Assert.assertEquals(description1, description2)
```
Comment (sherfert) [inline enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/ExecutionResultAcceptanceTest.scala]: Why are we using JUnit asserts in scala?

File: enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/ExecutionResultAcceptanceTest.scala [hunk 2]
```diff
@@ -38,4 +40,40 @@ class ExecutionResultAcceptanceTest extends ExecutionEngineFunSuite{
...
+    assert(description1.equals(description2))
+  }
+
+  test("without PROFILE it shouldn't be needed to iterate over the results before calling getExecutionPlanDescription in interpreted runtime") {
+    val query = "MATCH (n) WHERE n.prop = 1 RETURN n"
+    val runtime = InterpretedRuntimeOption.name
+
+    val description1 = executeQueryAndGetExecutionPlanDescription(query, runtime, false)
+    val description2 = executeQueryAndGetExecutionPlanDescription(query, runtime, true)
+    assert(description1.equals(description2))
```
Comment (pontusmelke) [inline enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/ExecutionResultAcceptanceTest.scala]: We usually prefer

```
description1 should equal(description2)
```

File: enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/ExecutionResultAcceptanceTest.scala [hunk 3]
```diff
@@ -38,4 +40,40 @@ class ExecutionResultAcceptanceTest extends ExecutionEngineFunSuite{
...
+  }
+
+  test("without PROFILE it shouldn't be needed to iterate over the results before calling getExecutionPlanDescription in slotted runtime") {
+    val query = "MATCH (n) WHERE n.prop = 1 RETURN n"
+    val runtime = SlottedRuntimeOption.name
+
+    val description1 = executeQueryAndGetExecutionPlanDescription(query, runtime, false)
+    val description2 = executeQueryAndGetExecutionPlanDescription(query, runtime, true)
+    assert(description1.equals(description2))
```
Comment (pontusmelke) [inline enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/ExecutionResultAcceptanceTest.scala]: same here use `a should equal(b)` instead

File: enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/ExecutionResultAcceptanceTest.scala [hunk 4]
```diff
@@ -38,4 +40,40 @@ class ExecutionResultAcceptanceTest extends ExecutionEngineFunSuite{
...
+  }
+
+  private def executeQueryAndGetExecutionPlanDescription(query: String, runtime: String, iterateOverResult: Boolean) = {
+    val executedQuery = "CYPHER runtime = " + runtime + " " + query
+    val result: Result = graph.execute(executedQuery)
+    if (iterateOverResult)
+      assert("empty iterator".equals(result.toString)) // don't really care for the assertion, just consume the results
```
Comment (pontusmelke) [inline enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/ExecutionResultAcceptanceTest.scala]: `result.hasNext should be false`

**Review by pontusmelke — CHANGES_REQUESTED** [review]
Just some small things

**Pull request merged**